### ISSUE TITLE
Store does not put closure around inherited transitions

### ISSUE THREAD

**Issue by brandynbennett** [microstates/microstates.js#287]
When creating a microstate that extends another microstate via the `Store` the `this` context does not reference transition methods inherited from a parent.  

Given
```js
export default class Person {
  firstName = String;
  lastName = String;

  get fullName() {
    return `${this.firstName.state} ${this.lastName.state}`;
  }

  changeIdentity(firstName, lastName) {
    return this
      .firstName.set(firstName)
      .lastName.set(lastName);
  }
}

export default class Infant extends Person {
  get fullName() {
    return `Baby ${this.firstName.state} ${this.lastName.state}`;
  }
}

let infant = Store(create(Infant), nextState => this.set('infant', nextState));
```

When you try and call `infant.changeIdentity` the `this` context may reference a component's context instead of the microstates.

Reproduction of the failure can be seen here: https://github.com/brandynbennett/microstates-test

Also see discussion on https://github.com/microstates/ember/issues/78

![kapture 2018-11-30 at 15 51 05](https://user-images.githubusercontent.com/4146629/49319611-df2a8400-f4ba-11e8-98a2-79dbf5747dc9.gif)

**Comment by taras** [comment]
It looks like it works on Node 10 but not on Chrome. I'll hook up Chrome to our test suite in Microstates and see why this is failing. 

Here is a reproduction of it in Node https://runkit.com/taras/microsates---descendant-class-bug

**Comment by taras** [comment]
To reproduce this error, I had to destructure the transition from the store object. I updated the runkit example with the failing case https://runkit.com/taras/microsates---descendant-class-bug

**Comment by taras** [comment]
Released in 0.12.3

**Comment by brandynbennett** [comment]
@taras This totally fixed it. Thanks for the fix!
